File: Assets/Common/Tools/FilePathAttribute.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UnityEngine;

namespace Assets.Common.Tools
{
    [AttributeUsage(AttributeTargets.Field)]
    public sealed class FilePathAttribute : PropertyAttribute
    {
        public string[] arrExt { get; private set; }
        public List<string> arrPath { get; private set; }
        public FilePathAttribute(string[] arrPath, string[] arrExt)
        {
            this.arrExt = arrExt;
            this.arrPath = arrPath.ToList<string>();
            if (this.arrPath.Count <= 0)
            {
                this.arrPath.Add("");
            }
        }
        public FilePathAttribute(string path, string ext)
        {
            this.arrExt = new string[] { ext };
            this.arrPath = new List<string>();
            this.arrPath.Add(path);
        }

        public FilePathAttribute(string ext)
        {
            this.arrExt = new string[] { ext };
            this.arrPath = new List<string>();
            this.arrPath.Add("");
        }

        public FilePathAttribute(string[] arrExt)
        {
            this.arrExt = arrExt;
            this.arrPath = new List<string>();
            this.arrPath.Add("");
        }
    }
}

File: Assets/.lpproj/Apis/Unity/CS/Assets/Common/Tools/FilePathAttribute.lua
---@class CS.Assets.Common.Tools.FilePathAttribute : CS.UnityEngine.PropertyAttribute
CS.Assets.Common.Tools.FilePathAttribute = {}

---@property readwrite CS.Assets.Common.Tools.FilePathAttribute.arrExt : CS.System.String[]
CS.Assets.Common.Tools.FilePathAttribute.arrExt = nil

---@property readwrite CS.Assets.Common.Tools.FilePathAttribute.arrPath : CS.System.Collections.Generic.List
CS.Assets.Common.Tools.FilePathAttribute.arrPath = nil

---@param arrPath : CS.System.String[]
---@param arrExt : CS.System.String[]
---@return CS.Assets.Common.Tools.FilePathAttribute
function CS.Assets.Common.Tools.FilePathAttribute(arrPath, arrExt)
end

---@param path : CS.System.String
---@param ext : CS.System.String
---@return CS.Assets.Common.Tools.FilePathAttribute
function CS.Assets.Common.Tools.FilePathAttribute(path, ext)
end

---@param ext : CS.System.String
---@return CS.Assets.Common.Tools.FilePathAttribute
function CS.Assets.Common.Tools.FilePathAttribute(ext)
end

---@param arrExt : CS.System.String[]
---@return CS.Assets.Common.Tools.FilePathAttribute
function CS.Assets.Common.Tools.FilePathAttribute(arrExt)
end